### PR TITLE
New partial-line test is flaking

### PR DESCRIPTION
**Pull request by edsantiago**
...add a 'sleep 1' in hopes of clearing the flake.

```release-note
None
```

### CODE REVIEW

**Comment by edsantiago** [comment]
Sample [flake](https://api.cirrus-ci.com/v1/artifact/task/6536551153270784/html/sys-podman-debian-13-rootless-host-sqlite.log.html#t--00127):
```
<+018ms> # $ podman logs --tail 1 tail_container
<+090ms> # Error: unable to read log file .../ctr.log : '2023-08-09T22:16:13.299101545+00:00' is not a valid container log line
```

File: test/system/035-logs.bats [hunk 1]
```diff
@@ -394,6 +394,8 @@ $content--2.*" "logs --until -f on running container works"
 
     # "-t" gives us ^Ms (CRs) in the log
     run_podman run --name $cname --log-driver k8s-file -t $IMAGE echo hi
+    # Sigh.... we can't even trust k8s-file. The log file gets written asynchronously.
```
Comment (vrothberg) [inline test/system/035-logs.bats:397]: I am still on my first coffee but that doesn't seem right.

The k8s-file gets written by Podman, so on exit of the process the log _must_ be written.  Maybe it's just the comment that confuses me.

Comment (Luap99) [reply test/system/035-logs.bats:397]: The file is written by conmon but I agree it should fully be written by the time podman run exits as this should wait for conmon to exit.

Second the test overwrites the file anyway so I do not see how --tail could end up with a partially written log line.

Comment (edsantiago) [reply test/system/035-logs.bats]: My suspicion (unproven) was that conmon was lingering with an open FD. But even if it was, the close() would/should have the file pointer at a place that would not leave an incomplete line.

Anyhow, I just re-pushed with a better approach: `rm` the file before writing to it. That should eliminate the possibility of multiple writers.

**Comment by edsantiago** [comment]
@lsm5 are you aware that pack-it is failing again?

**Comment by lsm5** [comment]
> @lsm5 are you aware that pack-it is failing again?

I'm aware now :D . Looks like dnf / gpg issues to me, should go away soon.

**Comment by lsm5** [comment]
> > @lsm5 are you aware that pack-it is failing again?
> 
> I'm aware now :D . Looks like dnf / gpg issues to me, should go away soon.

ohh btw, they've been branching fedora 39 from rawhide recently, so I suspect that might be the cause behind this.

**Comment by lsm5** [comment]
@edsantiago @cevich I'm thinking we could create some label to ignore packit failures. Checking with packit team if they have something of that sort already.

**Comment by lsm5** [comment]
> @edsantiago @cevich I'm thinking we could create some label to ignore packit failures. Checking with packit team if they have something of that sort already.

@edsantiago packit doesn't have anything of that sort atm. So, we can just `manually ignore` :D this for now and move on.

**Comment by edsantiago** [comment]
/hold

Still failing.

**Comment by lsm5** [comment]
> Still failing.

We can ignore the rawhide failure for now. It looks like a copr env issue. One other person has also reported it in the fedora copr matrix room.

**Comment by edsantiago** [comment]
No, I mean, failing IRL. In CI. As in, this PR does not fix anything.

(Sorry Lokesh, but we're all accustomed to ignoring packit failures)

**Comment by Luap99** [comment]
Do you have a reproducer? I have a hard time understanding the error condition here.
The error is saying `2023-08-09T22:16:13.299101545+00:00` is the only thing on the log line which means something in the log must been cut off (now that makes zero sense because we know we write a valid log).
Now the only other logical explanation is my changed code is incorrect and it for some reason cuts of in the middle of the line. The problem with that is I do not see what that would be flaky.

**Comment by edsantiago** [comment]
My reproducer is:
```console
$ echo '2023-08-10T08:48:45.298918348-06:00' >| ~/.local/share/..../ctr.log
```
I am of course trying a more realistic one so I can see what the log looks like. No luck yet.

**Comment by Luap99** [comment]
@edsantiago Assuming you see this in no flake retry PR feel free to apply this:
```diff
diff --git a/libpod/logs/log.go b/libpod/logs/log.go
index 4229501cf..574463455 100644
--- a/libpod/logs/log.go
+++ b/libpod/logs/log.go
@@ -83,6 +83,17 @@ func GetLogFile(path string, options *LogOptions) (*tail.Tail, []*LogLine, error
        return t, logTail, err
 }
 
+func printFileToStderr(path string) {
+       f, err := os.Open(path)
+       if err != nil {
+               logrus.Errorf("failed to read file: %v", err)
+       }
+       _, err = io.Copy(os.Stderr, f)
+       if err != nil {
+               logrus.Errorf("failed to print file: %v", err)
+       }
+}
+
 func getTailLog(path string, tail int) ([]*LogLine, error) {
        var (
                nllCounter int
@@ -120,7 +131,8 @@ func getTailLog(path string, tail int) ([]*LogLine, error) {
                        }
                        nll, err := NewLogLine(lines[i])
                        if err != nil {
-                               return nil, err
+                               printFileToStderr(path)
+                               return nil, fmt.Errorf("parse log line: %w", err)
                        }
                        if !nll.Partial() || first {
                                nllCounter++
@@ -146,7 +158,8 @@ func getTailLog(path string, tail int) ([]*LogLine, error) {
                        if leftover != "" && nllCounter < tail {
                                nll, err := NewLogLine(leftover)
                                if err != nil {
-                                       return nil, err
+                                       printFileToStderr(path)
+                                       return nil, fmt.Errorf("parse leftover log line: %w", err)
                                }
                                tailLog = append(tailLog, nll)
                        }
```

It should print the full file content to stderr and show me which of the two code paths failed.

**Comment by edsantiago** [comment]
@Luap99 I'd love to, but patch doesn't apply. I don't have 4229501cf in my repo. Is it possible that your patch is applied on other uncommitted local work in your repo? I can't even figure out how to apply the patch by hand.

**Comment by edsantiago** [comment]
Never mind, I hand-picked it (still don't know why it failed) and have pushed to #17831.

**Comment by lsm5** [comment]
@edsantiago looks like the copr folks fixed rawhide.

**Comment by edsantiago** [comment]
[Weird.....](https://api.cirrus-ci.com/v1/artifact/task/6548297570058240/html/sys-remote-debian-13-root-host-sqlite.log.html#t--00127)
```
<+008ms> # # podman-remote --url unix:/tmp/podman_tmp_oCoc inspect --format {{.HostConfig.LogConfig.Path}} tail_container
<+047ms> # /var/lib/containers/storage/overlay-containers/8b2990f1029b8a9789cc4ed6314f4b69c4251241fe89b724a5a96e158c01b67d/userdata/ctr.log
         # 2023-08-10T20:36:55.215741863+00:00
         # 2023-08-10T20:36:55.215856820+00:00 stdout F podman1

         # 2023-08-10T20:36:55.215741863+00:00
         # 2023-08-10T20:36:55.215856820+00:00 stdout P podman2
         # 2023-08-10T20:36:55.215741863+00:00
         # 2023-08-10T20:36:55.215856820+00:00 stdout F 

         # 2023-08-10T20:36:55.215741863+00:00
         # 2023-08-10T20:36:55.215856820+00:00 stdout F podman3

<+020ms> # # podman-remote --url unix:/tmp/podman_tmp_oCoc logs --tail 1 tail_container
<+041ms> # Error: failed to obtain logs for Container 'tail_container': unable to read log file 8b2990f1029b8a9789cc4ed6314f4b69c4251241fe89b724a5a96e158c01b67d for /var/lib/containers/storage/overlay-containers/8b2990f1029b8a9789cc4ed6314f4b69c4251241fe89b724a5a96e158c01b67d/userdata/ctr.log : parse log line: '2023-08-10T20:36:55.215741863+00:00' is not a valid container log line
```
Now I wonder if it's my `awk`? Anyhow, I'm really sure this is my fault, a test bug, not a bug in your (@Luap99's) code

**Comment by Luap99** [comment]
Right if the original log already contains two lines (partial + full) then the awk would return two timestamps thus corrupt the log. So I think we managed to add a test which runs into the same flake my PR fixed (incorrect handling of partial lines)  :laughing:

Do you really care about the current ts used? We could just hard code the ts and safe that awk step.

**Comment by edsantiago** [comment]
What I don't understand is how `echo hi` could generate a partial line, but that's something I'll worry about later. Or never. 

I added `head -1`, and ran three test cycles on it without any flakes. I don't have strong feelings about `awk` vs a hardcoded TS, if you or someone prefer hardcoded I will update the PR next week. Otherwise I think it's ready for review.

**Comment by Luap99** [comment]
I am good with it either way, given tests are green LGTM. It is not worth to repush for that.

**Comment by edsantiago** [comment]
On reflection I have a mild preference for keeping it this way, for separation-of-knowledge reasons: with this form, the test has to know `<timestamp> <P|F> stdout <msg>` but does not need to know what `timestamp` looks like. And FWIW I've given this 7-8 spins in #17831 with no failures.

**Pull request merged**